docs/dg/dev/guidelines/performance-guidelines/database-performance-quidelines.md

```diff
@@ -0,0 +1,155 @@
+
+### Usage example
+
+```php
+class AbstractSpySalesOrderItem extends BaseSpySalesOrderItem implements BatchEntityPostSaveInterface
+{
+    // properties re-declared protected to prevent their direct use out from an entity
+    use CascadeActiveRecordBatchProcessorTrait {
+        persist as protected;
+        commit as protected;
+        commitIdentical as protected;
+    }
+    public function batchPostSave(): void
+    {
+        if ($this->statusChanged && $this->getIdSalesOrderItem()) {
+            /** @var \Orm\Zed\Sales\Persistence\SpySalesOrderItem $salesOrderItemEntity */
+            $salesOrderItemEntity = $this;
+            $omsOrderItemStateHistoryEntity = $this->createOmsOrderItemStateHistoryEntity();
+            $omsOrderItemStateHistoryEntity->setOrderItem($salesOrderItemEntity);
+            $omsOrderItemStateHistoryEntity->setFkOmsOrderItemState($this->getFkOmsOrderItemState());
+
+            $this->sharedPersist($omsOrderItemStateHistoryEntity);
+        }
+
+        $this->statusChanged = false;
+    }
+}
+```
+
+With this implementation, the following code:
+
+```php
+class SalesEntityManager extends AbstractEntityManager implements SalesEntityManagerInterface
+{
+    public function updateOrCreateSalesOrderItems(array $salesOrderItemsData): void
+    {
+        // Use batch processor trait for efficient processing
+        foreach ($salesOrderItemsData as $orderItemData) {
+            // Find or create the sales order item
+            $salesOrderItem = $this->findOrCreateSalesOrderItem($orderItemData);
+    
+            // Update item attributes
+            $this->updateSalesOrderItemAttributes($salesOrderItem, $orderItemData);
+    
+            // Collect for batch processing
+            $this->persist($salesOrderItem);
+        }
+    
+        // Commit all collected items in a single batch operation
+    $this->commit();
+    }
+}
+```
+
+Performs the following actions:
+- Loads all required entities from the database in a single query.
+- Creates or updates `salesOrderItem` entities in a single batch.
+- Creates related `OrderItemStateHistoryEntity` instances during the post-save process execution (method `batchPostSave` declared by `BatchEntityPostSaveInterface`) and saves them in a single operation after the main entity save is complete.
+
+## BatchEntityHooksInterface
+
+BatchEntityHooksInterface enables `postSave` hooks, which are similar to publish and synchronize (P&S) events.
+
+To ensure proper event handling during batch operations, the corresponding interface must be implemented for storage and search entities when used with `ActiveRecordBatchProcessorTrait` or `CascadeActiveRecordBatchProcessorTrait`. These traits don't automatically trigger P&S events for storage or search entities. Therefore, explicit implementation is required to handle such events correctly during batch processing.
```

Comment:
what do you mean by the 'corresponding interface'?

Reply:
The topic there is `BatchEntityHooksInterface`, i mean that interface. Should it be mentioned?